ast/functions_test.go

```diff
@@ -227,3 +227,19 @@ func (ts *testDMLSuite) TestWindowFuncExprRestore(c *C) {
 	}
 	RunNodeRestoreTest(c, testCases, "select %s from t", extractNodeFunc)
 }
+
+func (ts *testFunctionsSuite) TestGenericFuncRestore(c *C) {
+	testCases := []NodeRestoreTestCase{
+		{"s.a()", "`s`.`a`()"},
+		{"`s`.`a`()", "`s`.`a`()"},
+		{"now()", "NOW()"},
+		{"`s`.`now`()", "`s`.`now`()"},
+		// FIXME: expectSQL should be `generic_func()`.
+		{"generic_func()", "GENERIC_FUNC()"},
```

Comment:
To identify these functions(without a dot), the first branch of `FunctionCallGeneric` rule needs to be changed from `identifier` to `Identifier`, and use `CheckInTokenMap` to filter generic functions. 
```
FunctionCallGeneric:
	identifier '(' ExpressionListOpt ')'		identifier '(' ExpressionListOpt ')'
```
However, such change also causes a lot of shift/reduce and reduce/reduce conflicts (with FunctionCallKeyword and FunctionCallNonKeyword).

Reply:
One more thing: hundreds of test cases still need to be updated even if the conflicts are resolved.

Reply:
🤔 That's gonna break a lot of test cases.